tests/model_formsets/test_uuid.py

```diff
@@ -85,9 +85,34 @@ def test_inlineformset_factory_nulls_default_pks_alternate_key_relation(self):
         #24958 - Variant of test_inlineformset_factory_nulls_default_pks for
         the case of a parent object with a UUID alternate key and a child
         object that relates to that alternate key.
+        #32210 - Parent object with a UUID alternate key should not None.
         """
         FormSet = inlineformset_factory(
             ParentWithUUIDAlternateKey, ChildRelatedViaAK, fields="__all__"
         )
         formset = FormSet()
-        self.assertIsNone(formset.forms[0].fields["parent"].initial)
+        self.assertIsNotNone(formset.forms[0].fields["parent"].initial)
```

Comment:
I have to admit that this still feels a bit odd to me 🤔 
Especially since the tests above are expecting the initial value to be None, and we are perhaps lacking a bit of explanation why this is the case here.

Reply:
For this case, When the Parent model has a UUID field but that's not the primary key and has a default value `uuid.uuid4`, so UUID field becomes None due to some code in the models.py file, and occur IntegrityError on save data.

Reply:
Yep, and I think that this behaviour is not obvious.
What do you think of adding a comment to explain exactly this?

Reply:
@David-Wobrock 
I did comment on the test and explained why I used the NotNone case to check the initial value.